omod/src/main/webapp/fragments/patientdashboard/visits.gsp

```diff
@@ -70,8 +70,15 @@
     <ul id="visits-list" class="left-menu">
         <%
             def visits = patient.allVisitsUsingWrappers;
+
             visits.eachWithIndex { wrapper, idx ->
-                def primaryDiagnoses = wrapper.getUniqueDiagnoses(true, false)
+            def primaryDiagnoses = wrapper.getUniqueDiagnoses(true, false);
+           
+            for (def encounter : wrapper.getSortedEncounters()){
+               def  newdiagnoses = diagnosisService.getPrimaryDiagnoses(encounter);
+               primaryDiagnoses << [newdiagnoses]
```

Comment:
@reagan-meant, I guess this works best with ordinary arrays than with lists. I would suppose you do something like `primaryDiagnoses.addAll(newdiagnoses)`.

Reply:
You are right...I came to the same conclusion after checking up on the different flavours of <<, addAll, "+" and push that groovy offers.....Will be making an updated PR